### PR TITLE
Test dynamic flags v2

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr

### CODE REVIEW

File: tests/framework/e2e/cluster.go [hunk 1]
```diff
@@ -509,27 +509,27 @@ func (cfg *EtcdProcessClusterConfig) EtcdServerProcessConfig(tb testing.TB, i in
...
 	args := []string{
-		"--name", name,
-		"--listen-client-urls", strings.Join(curls, ","),
-		"--advertise-client-urls", strings.Join(curls, ","),
-		"--listen-peer-urls", peerListenUrl.String(),
-		"--initial-advertise-peer-urls", peerAdvertiseUrl.String(),
-		"--initial-cluster-token", cfg.ServerConfig.InitialClusterToken,
-		"--data-dir", dataDirPath,
-		"--snapshot-count", fmt.Sprintf("%d", cfg.ServerConfig.SnapshotCount),
+		"--name=" + name,
```
Comment (ahrtr) [inline tests/framework/e2e/cluster.go:512]: a quick question: why change this? do you see any issue?

Comment (ahrtr) [reply tests/framework/e2e/cluster.go:512]: it seems that you just want to make it easier to verify the args in UT?

**Comment by ahrtr** [comment]
Please ping me when the test is green.

File: tests/framework/e2e/cluster.go (hunk 1 above)
Comment (serathius) [reply tests/framework/e2e/cluster.go:512]: Yes, I want to be 100% sure that switching to dynamically generated flags doesn't make any changes in the provided flags.

**Comment by serathius** [comment]
Ups, looks like a surprising bug. I was not able to switch `--data-dir` to use `=`. Don't know why, might require investigation as this looks like a bug in etcd server.

**Comment by serathius** [comment]
ping @ahrtr

**Comment by ahrtr** [comment]
> I was not able to switch `--data-dir` to use `=`. Don't know why, might require investigation as this looks like a bug in etcd server.

I assume you will create a followup.

**Pull request merged**